btcec/schnorr/musig2/musig2_test.go

```diff
@@ -82,34 +119,98 @@ func TestMuSig2KeyAggTestVectors(t *testing.T) {
 			keyOrder:    []int{0, 0, 1, 1},
 			expectedKey: keyCombo4,
 		},
+
+		// Invalid public key
+		{
+			keyOrder:      []int{0, 3},
+			expectedError: secp256k1.ErrPubKeyNotOnCurve,
+		},
+
+		// Public key exceeds field size
+		{
+			keyOrder:      []int{0, 4},
+			expectedError: secp256k1.ErrPubKeyXTooBig,
+		},
+
+		//  Tweak is out of range
+		{
+			keyOrder: []int{0, 1},
+			tweaks: []KeyTweakDesc{
+				KeyTweakDesc{
+					Tweak:   to32ByteSlice(invalidTweak),
+					IsXOnly: true,
+				},
+			},
+			expectedError: ErrTweakedKeyOverflows,
+		},
+
+		// Intermediate tweaking result is point at infinity
+		{
+			explicitKeys: []*secp256k1.PublicKey{GetGenerator()},
+			tweaks: []KeyTweakDesc{
+				getInfinityTweak(),
+			},
+			expectedError: ErrTweakedKeyIsInfinity,
+		},
 	}
 	for i, testCase := range testCases {
 		testName := fmt.Sprintf("%v", testCase.keyOrder)
 		t.Run(testName, func(t *testing.T) {
 			var (
-				keys    []*btcec.PublicKey
-				strKeys []string
+				keys      []*btcec.PublicKey
+				strKeys   []string
+				strTweaks []jsonTweak
+				jsonError string
 			)
 			for _, keyIndex := range testCase.keyOrder {
 				keyBytes := testKeys[keyIndex]
 				pub, err := schnorr.ParsePubKey(keyBytes)
 				if err != nil {
+					if errors.Is(err, testCase.expectedError) {
+						return
```

Comment:
Why return vs continue here? IIUC, we want to continue to do all the cases after the first invalid one. 

Another way to write this would be:
```go
switch {
    case testCase.expectedError != nil && errors.Is(err, testCase.expectedError)
        continue:
    case err != nil:
        t.Fatalf()
}

Reply:
Do you mean that we want to test all the remaining keys of the current testcase? This returns from ```t.Run(testname, func(t *testing){// returns from here})``` and moves to the next testcase.